pkg/sdk/functions_ext.go

```diff
@@ -69,9 +92,136 @@ func functionDetailsFromRows(rows []FunctionDetail) (*FunctionDetails, error) {
 			v.TargetPath = row.Value
 		}
 	}
+	if e := errors.Join(errs...); e != nil {
+		return nil, e
+	}
+
+	if functionDetailsImports, err := parseFunctionDetailsImport(*v); err != nil {
+		errs = append(errs, err)
+	} else {
+		v.NormalizedImports = functionDetailsImports
+	}
+
+	if v.TargetPath != nil {
+		if p, err := parseStageLocationPath(*v.TargetPath); err != nil {
+			errs = append(errs, err)
+		} else {
+			v.NormalizedTargetPath = p
+		}
+	}
+
+	if dt, returnNotNull, err := parseFunctionOrProcedureReturns(v.Returns); err != nil {
+		errs = append(errs, err)
+	} else {
+		v.ReturnDataType = dt
+		v.ReturnNotNull = returnNotNull
+	}
+
+	if args, err := parseFunctionOrProcedureSignature(v.Signature); err != nil {
+		errs = append(errs, err)
+	} else {
+		v.NormalizedArguments = args
+	}
+
 	return v, errors.Join(errs...)
 }
 
+// TODO [SNOW-1850370]: use ParseCommaSeparatedStringArray + collections.MapErr combo here and in other methods?
+func parseFunctionDetailsImport(details FunctionDetails) ([]NormalizedPath, error) {
+	functionDetailsImports := make([]NormalizedPath, 0)
+	if details.Imports == nil || *details.Imports == "" || *details.Imports == "[]" {
+		return functionDetailsImports, nil
+	}
+	if !strings.HasPrefix(*details.Imports, "[") || !strings.HasSuffix(*details.Imports, "]") {
+		return functionDetailsImports, fmt.Errorf("could not parse imports from Snowflake: %s, wrapping brackets not found", *details.Imports)
+	}
+	raw := (*details.Imports)[1 : len(*details.Imports)-1]
+	imports := strings.Split(raw, ",")
+	for _, imp := range imports {
+		p, err := parseStageLocationPath(imp)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse imports from Snowflake: %s, err: %w", *details.Imports, err)
+		}
+		functionDetailsImports = append(functionDetailsImports, *p)
+	}
+	return functionDetailsImports, nil
+}
+
+func parseStageLocationPath(location string) (*NormalizedPath, error) {
+	log.Printf("[DEBUG] parsing stage location path part: %s", location)
+	idx := strings.Index(location, "/")
+	if idx < 0 {
+		return nil, fmt.Errorf("part %s cannot be split into stage and path", location)
+	}
+	stageRaw := strings.TrimPrefix(strings.TrimSpace(location[:idx]), "@")
```

Comment:
What about stages with `/` in their name? If they're not supported, let's mention it in the docs.